### PR TITLE
Preventing duplicate entries from being entered into the sql_event_store

### PR DESCRIPTION
**Pull request by gzusaf**

### CODE REVIEW

File: task_queue/events/in_memory_event_store.py [hunk 1]
```diff
@@ -23,11 +23,19 @@ def _add_raw(self, events):
...
             if event.name not in self.events:
                 self.events[event.name] = []
-
-            event.id = len(self.events[event.name])
-            self.events[event.name].append(event)
+                event.id = len(self.events[event.name])
+                self.events[event.name].append(event)
+            else:
+                for e in self.events[event.name]:
+                    if e.name == event.name and e.data == event.data:
```
Comment (AM-USAF) [inline task_queue/events/in_memory_event_store.py:33]: Is this necessary because there may be an instance where two events have the same name but different data and that is allowed?

Comment (CK-USAF) [reply task_queue/events/in_memory_event_store.py:33]: I think this check is if both the name and data are the same then it catches the duplicate

File: task_queue/events/sql_event_store.py [hunk 2]
```diff
@@ -84,8 +100,12 @@ def _add_raw(self, events):
...
         ]
 
-        for db_evt in db_events:
-            print(db_evt)
+        for db_evt in db_events[:]:
+            if self._check_for_duplicates(db_evt):
+                db_events.remove(db_evt)
+
+        if not db_events:
+            return
```
Comment (AM-USAF) [inline task_queue/events/sql_event_store.py:108]: Why is this needed? Is it the same reason as the return on line 96?

**Review by AM-USAF — DISMISSED** [review]
My comments aren't blocking, just wondering why things are the way they are.

File: task_queue/events/in_memory_event_store.py (hunk 1 above)
Comment (AM-USAF) [reply task_queue/events/in_memory_event_store.py:33]: yes I can see what it was doing I was wondering if it was necessary, I don't know enough about Events to know if it is possible to have the same event name and different data so that's why I asked if that was allowed

Comment (gzusaf) [reply task_queue/events/in_memory_event_store.py:33]: It is necessary. In Orange Strike's usage of the queue the same event can be applied to large amounts of data so there will many Events that share the same `event_name`.
`event.data` contains the unique information about the data that the event is being applied to.

File: task_queue/events/sql_event_store.py (hunk 2 above)
Comment (gzusaf) [reply task_queue/events/sql_event_store.py:108]: Yes it's the same empty list check as before.
This is needed in the case where the entire db_events list contains only duplicates.

**Review by BE-USAF — APPROVED** [review]

**Review by AB-USAF — APPROVED** [review]

File: task_queue/events/in_memory_event_store.py [hunk 3]
```diff
@@ -23,11 +23,19 @@ def _add_raw(self, events):
             List of Events
         """
         for event in events:
+            duplicate = False
             if event.name not in self.events:
                 self.events[event.name] = []
-
-            event.id = len(self.events[event.name])
-            self.events[event.name].append(event)
+                event.id = len(self.events[event.name])
```
[the remark this answers is not in the document]
Comment (BE-USAF) [reply task_queue/events/in_memory_event_store.py:29]: Do you wanna ping @gzusaf directly?

[the remark this answers is not in the document]
Comment (BE-USAF) [reply task_queue/events/in_memory_event_store.py:29]: I believe it's just to be stored in the event object to be used later

**Review by JR-USAF — APPROVED** [review]
